### ISSUE TITLE
Implement censored data

### ISSUE THREAD

**Issue by ibab** [ibab/python-mle#26]
When only a certain range of the data is used for the fit, or when a range has been cut out, the PDFs need to be adjusted accordingly by removing the missing parts and renormalizing.

I'm thinking of something like this for the API
```python
x = var('x', observed=True, vector=True)
mu = var('mu')
sigma = var('sigma')
model = Censor(x > 0, Normal(x, mu, sigma))
```

**Comment by KonstantinSchubert** [comment]
I don't think that 'censor' is a good keyword here, it makes sense only in a certain use case. 

Something like "Range" would be more fitting. And I think it should be a feature of the variable.
